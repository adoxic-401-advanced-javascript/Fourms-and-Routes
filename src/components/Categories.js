import React from 'react';
import Characters from './Characters';
import fetchXfile from '../utils/fetch';

export default function Categories(category) {
  const [state, setState] = React.useState([]);

  React.useEffect(() => {
    handleClick();
  }, []);

  const handleClick = () => {
    if(category.category === 'Main characters' || category.category === 'Monster of the Week' || category.category === 'Syndicate' || category.category === 'Informants') {
      fetchXfile(`https://xfiles-api.herokuapp.com/api/v1/characters?category=${category.category}`)
        .then(res => {
          setState(res.results);
        });

    } else {
      fetchXfile(`https://xfiles-api.herokuapp.com/api/v1/characters/${category.category}`)
        .then(res => {
          setState(res);
        });
    }
  };

  return (
    <>
      <h2>{category.category}</h2>
      <Characters title={state} characters={state} />
    </>
  );
}
