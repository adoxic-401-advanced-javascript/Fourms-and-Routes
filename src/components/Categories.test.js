import React from 'react';
import { shallow } from 'enzyme';
import Categories from './Categories';

describe('Categories Test', () => {
  it('Renders an Categories', () => {
    const wrapper = shallow(<Categories 
      title={'title'}
      characters={['thing']}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
