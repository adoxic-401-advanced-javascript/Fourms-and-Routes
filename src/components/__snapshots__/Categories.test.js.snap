// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Test Renders an Categories 1`] = `
<Fragment>
  <h2 />
  <Characters
    characters={Array []}
    title={Array []}
  />
</Fragment>
`;
